src/app/hooks/useTimerEffect.ts

```diff
@@ -0,0 +1,13 @@
+import { DependencyList, useEffect } from 'react';
+
+import { EmptyFn } from '../config/general';
+
+export const useTimerEffect = (callback: EmptyFn, refreshInterval: number, deps: DependencyList = []) =>
+  useEffect(() => {
+    callback();
+
+    const interval = setInterval(callback, refreshInterval);
+
+    return () => clearInterval(interval);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [...deps]);
```

Comment:
use `deps` without the spread operator to omit this es-lint warning.  
and remove `deps` default value (let `useTimerEffect` have the same api as `useEffect` have)